### ISSUE TITLE
Preprocessor: BinaryExpression is not correctly evaluate 

### ISSUE THREAD

**Issue by 7645re** [ForNeVeR/Cesium#521]
Test to reproduce the error:
```csharp
    [Fact]
    public Task Test() => DoTest(@"#define NUM 9
#if NUM < 10
int int() {}
#else
int main() {}
#endif
");
```

This line is executed incorrectly - `#if NUM < 10` - `False`

This is because in the EvaluateExpression method, the BinaryExpression class made an error comparing values. It uses a lexicographic comparison that compares strings character by character, so it turns out 9 is greater than 10, only 9 and 1 are compared

```csharp
return Operator switch
        {
            CPreprocessorOperator.Equals => firstValue == secondValue ? "1" : "0",
            CPreprocessorOperator.NotEquals => firstValue != secondValue ? "1" : "0",
            CPreprocessorOperator.LessOrEqual => (firstValue ?? "").CompareTo(secondValue ?? "") <= 0 ? "1" : "0",
            CPreprocessorOperator.GreaterOrEqual => (firstValue ?? "").CompareTo(secondValue ?? "") >= 0 ? "1" : "0",
            CPreprocessorOperator.LessThan => (firstValue ?? "").CompareTo(secondValue ?? "") < 0 ? "1" : "0",
            CPreprocessorOperator.GreaterThan => (firstValue ?? "").CompareTo(secondValue ?? "") > 0 ? "1" : "0",
            CPreprocessorOperator.LogicalAnd => (firstValue.AsBoolean() && secondValue.AsBoolean()) ? "1" : "0",
            CPreprocessorOperator.LogicalOr => (firstValue.AsBoolean() || secondValue.AsBoolean()) ? "1" : "0",
            _ => throw new CompilationException($"Operator {Operator} cannot be used in the preprocessor directives"),
        };
```

it would also be nice to write a handler for the entered numbers, because now it is possible to enter `#define NUM 00`

DOD:
- [ ] Check the other comparison operators
- [ ] Fix calculation errors
- [ ] Write tests for all possible cases 

Optional DOD:
- [ ] Number handler with input verification
- [ ] Handler tests

**Comment by 7645re** [comment]
⚔️My chivalrous soul wants to do it myself!

**Comment by ForNeVeR** [comment]
Closed by #531.
